[SanyaS] iP

sunpterodactyl opened the pull request:
# Sunpter

### Bonjour :wave: , I'm your go-to French chatbot 

Translations on the way!
> I know a few English words—just not forget. We won’t be needing that one here. 😉

### Why me
1. ~~Why not~~
2. Remember different types of tasks
3. Chat with me on the go
4. Simple to use

## Features:
- [x] Managing tasks
- [ ] Storage
- [ ] Chatbot Interface (coming soon!)

##
If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SSjrB committed:
Hardcoded a greet and exit message in Sunpter.java. Refactored all occurences of Duke to Sunpter

SSjrB committed:
Sunpter is now french. Added in scanner to echo input and made horizontalLines easier to use

SSjrB committed:
Added in a list in main file. Update and list functionalities.

SSjrB committed:
Created a new class to represent tasks

SSjrB committed:
Refactored String Array to Task array.

SSjrB committed:
Added in a roster class. Refactored my code. Also added in marking and unmarking capabilities

SSjrB committed:
Created the subclasses Deadline, ToDo, Event class

SSjrB committed:
Messy, but I added in a factory method to parse the input and create the respective task. Also refactored the add code in Supter.java

SSjrB committed:
Added in testing. Changed main file to ensure ACTUAL.txt aligned with EXPECTED.txt

SSjrB committed:
Added support for exceptions: entering a different variable type or not entering the right command formatting. It works for now

SSjrB committed:
Added in delete function and output message

SSjrB committed:
Cleanup

SSjrB committed:
Replaced task completion boolean with an enum

SSjrB committed:
Cleaned up the code and fully updated the enums function. Added error handling for empty/incorrect commands.

SSjrB committed:
Refactor: Implement Command pattern and enhance OOP principles

SSjrB committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support' into A-OOP
Add gradle support.

SSjrB committed:
Add test folder with 2 Junit task implementations for Parser and Storage classes.

SSjrB committed:
Add JavaDoc comments

SSjrB committed:
Implement in progress command

SSjrB committed:
Add search command

jonzyyyy commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,113 @@
+package command;
+//TODO: update exception handling
```
Maybe you forgot to update?

jonzyyyy commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,113 @@
+
+    /**
+     * Parses input and creates the task accordingly
+     * @param input
```
To leave a line break between the description and @param & @return comments

jonzyyyy commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,113 @@
+/**
+ * Command that takes in input and adds the given task to the taskList
+ */
+public class AddCommand extends Command {
```
This class name sounds more like a verb rather than a noun. Maybe commandAdder would be a better alternative?

jonzyyyy commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,28 @@
+/**
+ * Remove a task from the tasklist by citing its index number
+ */
+public class DeleteCommand extends Command {
```
Similarly, for this - should use a noun instead of a verb

jonzyyyy reviewed (commented):
Great work! Just some minor recommendations:)

delonlee01 commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,113 @@
+package command;
+//TODO: update exception handling
+import storage.Storage;
+import task.*;
```
Shouldn't this import be explicitly listed?

delonlee01 commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,113 @@
+    }
+
+    /**
+     * Execute function
```
Maybe this can be more descriptive?

delonlee01 commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,113 @@
+                ui.showTaskAddedMessage(task, roster.numberofTasks());
+            }
+        }
+        catch (NullPointerException e) {
```
Should this be on a newline?

delonlee01 commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,113 @@
+     */
+    private String[] getDeadline(String input, Ui ui, Pattern pattern) {
+        String item="";
+        String date="";
```
Shouldn't here have whitespaces before and after the `=` operator?

delonlee01 commented on src/main/java/command/InProgressCommand.java:
```diff
@@ -0,0 +1,23 @@
+    @Override
+    public void execute(Roster roster, Ui ui, Storage storage) {
+        roster.markTaskAsUncompleted(num);
+        ui.markAsUncompleted(roster,num);
```
Shouldn't here have whitespace after `,`? I noticed the same issue in other places too.

delonlee01 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,63 @@
+
+import command.*;
+import exception.SunpterException;
+import task.*;
```
Shouldn't these imports be explicitly listed?

delonlee01 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    /**
+    Parse the number of the task for commands such as mark, unmark, and delete.
```
```suggestion
    Parses the number of the task for commands such as mark, unmark, and delete.
```

delonlee01 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,41 @@
+    public Storage() {}
+
+    /**
+     * Save tasks into a file
```
```suggestion
     * Saves tasks into a file
```

delonlee01 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    /**
+     * Read saved tasks from the file sunpter.txt and loads it into the chatbot
```
I noticed the same issue in several places too.
```suggestion
     * Reads saved tasks from the file sunpter.txt and loads it into the chatbot
```

delonlee01 commented on src/main/java/task/Roster.java:
```diff
@@ -0,0 +1,103 @@
+/*
+This class helps keep encapsulate and update a list of tasks
+CRUD operations
+ */
```
Should this be here?

delonlee01 commented on src/main/java/task/Roster.java:
```diff
@@ -0,0 +1,103 @@
+     * Return the number of tasks in the roster
+     * @return int
+     */
+    public int numberofTasks() {
```
Shouldn't this be written with a verb in camelCase? e.g. `getNumberOfTasks`

delonlee01 reviewed (commented):
Looks like you're making good progress! 💪🏻

SSjrB committed:
Add in translations for Sunpter interactions

SSjrB committed:
Add GUI interface

SSjrB committed:
Fix UI configuration

SSjrB committed:
Change CSS styling and sunpter.png

SSjrB committed:
Add assertions to the mark and unmark commands

- assert that the task found at the given index is not null
- removed uneeded assertions in this file and replaced them with exceptions

SSjrB committed:
Add index out of bounds exception handling

- Added index out of bounds exceptions to Delete Mark, Unmark, and In progress commands
- Exception handling in Sunpter.java
- Removed instances of incorrect exception handling with SunpterException

SSjrB committed:
Remove unused code to implement java coding standard

SSjrB committed:
Add stream usage to the search operation in Roster

sunpterodactyl committed:
Update README.md

sunpterodactyl committed:
Merge pull request #2 from sunpterodactyl/branch-A-Assertions

Branch A assertions

SSjrB committed:
Merge branch 'master' into branch-A-CodeQuality
Merge master branch into branch-A-CodeQuality

sunpterodactyl committed:
Merge pull request #4 from sunpterodactyl/branch-A-CodeQuality

Branch a code quality

SSjrB committed:
Merge changes from branch-A-streams

sunpterodactyl committed:
Merge branch 'master' into branch-A-streams

sunpterodactyl committed:
Merge pull request #3 from sunpterodactyl/branch-A-streams

Add stream usage to the search operation in Roster

SSjrB committed:
Refactor functional call numberofTasks to numberOfTasks

SSjrB committed:
Merge branch 'branch-A-streams'

sunpterodactyl committed:
Create gradle.yml

Used pre-configured CI workflow for a Java application with gradle

sunpterodactyl committed:
Delete .github/workflows/ci-workflow.yml

SSjrB committed:
Add in treeset to handle sorting tasks by priority

- Read and populate a treeset from the current rosterlist
- Print out the tasks

SSjrB committed:
Add priority to string representation

SSjrB committed:
Clean up error handling in the parser

SSjrB committed:
Fix priority list and clean up error handling

sunpterodactyl committed:
Merge pull request #5 from sunpterodactyl/branch-BCD-extension

Merge branch bcd extension

SSjrB committed:
Show exit message before exiting the application

SSjrB committed:
Delay exit command

SSjrB committed:
Remove french to make the chatbot more user-friendly

SSjrB committed:
Add in priority progress message

SSjrB committed:
Fix sunpter response formatting

SSjrB committed:
Clean up DateTime parsing and exception handling

sunpterodactyl committed:
Update README.md

SSjrB committed:
Add product screenshot

sunpterodactyl committed:
Update gradle.yml

SSjrB committed:
Merge branch 'master' of https://github.com/sunpterodactyl/ip

sunpterodactyl committed:
Include product screenshot in README.md

SSjrB committed:
Fix storage loadTasks bug
- Store tasks in the storage in a separate storage format
- Parse tasks using the storageParser class
- Update priority points to update according to stored tasks

SSjrB committed:
Merge branch 'master' of https://github.com/sunpterodactyl/ip
Merge branch to master.

sunpterodactyl committed:
Add jar file

SSjrB committed:
Upload Ui.png file

sunpterodactyl committed:
Update README.md

SSjrB committed:
Change character to Will Smith

- Changed the Ui
- Fixed some error handling and messages
- Tweaked messages
- I dont know why I did this

SSjrB committed:
Merge branch 'master' of https://github.com/sunpterodactyl/ip
Change character to Will Smith.
- Just for fun

sunpterodactyl committed:
Update README.md

SSjrB committed:
Update tests
